applications/nrf_desktop/src/modules/led_stream.c

```diff
@@ -0,0 +1,163 @@
+    pos += sizeof(((struct led_color *)0)->c[0]);
+    steps_queue[position].substep_count=dyndata->data[pos];
+    pos += sizeof(((struct led_effect_step *)0)->substep_count);
+    steps_queue[position].substep_time=dyndata->data[pos];
+}
+
+static void send_effect(struct led_effect *effect)
+{
+    LOG_INF("Send_effect");
+    struct led_event *led_event = new_led_event();
+    led_event->stream = true;
+    led_event->led_id = STREAM_LED_ID;
+    led_event->led_effect=effect;
+
+    __ASSERT_NO_MSG(led_event->led_effect->steps);
+
+    EVENT_SUBMIT(led_event);
+}
+
+static void store_data(const struct event_dyndata *data)
+{
+    if(last_added_position < QUEUE_SIZE){
+        last_added_position++;
+        LOG_INF("Insert data on position %u", last_added_position);
+        queue_data(data, last_added_position);
+    } else {
+        LOG_INF("Queue is full");
+    }
+}
+
+static void send_data_from_queue()
+{
+    if(send_position <= last_added_position){
+        LOG_INF("Sending data %u, %u, %u from position %u", steps_queue[send_position].color.c[0],
+                steps_queue[send_position].color.c[1], steps_queue[send_position].color.c[2], send_position);
+        led_stream_effect.steps = &steps_queue[send_position];
+        send_effect(&led_stream_effect);
+        send_position++;
+    } else {
+        LOG_INF("No steps ready in queue");
+        last_added_position = -1;
+        send_position = 0;
+        first_sended = false;
+    }
+}
+
+static void handle_incoming_step(const struct config_event *event)
+{
+    store_data(&event->dyndata);
+    if (!first_sended){
+        LOG_INF("Sending first led effect");
+        send_data_from_queue();
+        first_sended = true;
+    }
+}
+
+static void handle_led_ready_event(const struct led_ready_event *event)
+{
+    LOG_INF("Led ready event handler");
+    send_data_from_queue();
+}
+
+static void handle_config_event(const struct config_event *event)
+{
+
+    if (!event->store_needed) {
+        /* Accept only events coming from transport. */
+        return;
+    }
+
+    if (GROUP_FIELD_GET(event->id) != EVENT_GROUP_LED_STREAM) {
+        /* Only LED STREAM events. */
+        return;
+    }
+
+    switch (TYPE_FIELD_GET(event->id)) {
+        case STREAM_DATA:
+            handle_incoming_step(event);
+            break;
+
+        default:
+            /* Ignore unknown event. */
+            LOG_WRN("Unknown LED STREAM config event");
+            break;
+    }
+
+}
+
+static bool event_handler(const struct event_header *eh)
+{
+    LOG_INF("Event handler");
+
+    if (IS_ENABLED(CONFIG_DESKTOP_CONFIG_CHANNEL_ENABLE)) {
+        if (is_config_event(eh)) {
+            handle_config_event(cast_config_event(eh));
+            return false;
+        }
+    }
+
+    if (is_led_ready_event(eh)) {
+        handle_led_ready_event(cast_led_ready_event(eh));
+        return false;
+    }
+
+    if (is_module_state_event(eh)) {
```

Comment:
Set MODULE_STATE_READY here - in reaction e.g. for led module being ready (we probably should not send events to `leds` before they are ready)

Reply:
done